doc/smart-switch/high-availability/smart-switch-ha-hld.md

```diff
@@ -0,0 +1,2176 @@
+
+## 2. Terminology
+
+| Term | Explanation |
+| ---- | ----------- |
+| ENI / vNIC / vPort | Elastic Network Interface. It means a VM’s NIC.<br/><https://github.com/sonic-net/DASH/wiki/Glossary#eni> |
+| T0/1/2 | Tier 0/1/2 switch. |
+| Session | Session represents a network connection between 2 sides of the network, for example, TCP or UDP. |
+| Flow | Each session has 2 flows in network – Forwarding flow and reverse flow for process the network packet in both directions. |
+| HA | High Availability |
+| HA Set | The set of DPUs that involved in providing the high availability for certain HA scope – switch or DPU or ENI. |
+| HA Scope | The minimum scope that failover can happen on, for example – entire switch, entire DPU or each ENI. Every HA scope tracks its own active instance, and others being standby.  |
+| HA Active | The active instance in the HA set. It is the only instance in the HA set that can decide what the packet transformation we should apply any every new flow. |
+| HA Standby | The standby instance in the HA set. It acts like a flow store and never makes any decisions. If any new connections happen to land on the standby instance, it will be either tunneled to the active. |
+| DP | Data path. |
+| DP VIP | Data path Virtual IP. This is the VIP that SmartSwitch advertises via BGP for attracting traffic.<br/><br/>After enabling SmartSwitch, when VM sends out traffic, it will be encap’ed with the destination IP set to the DP VIP. This makes all the VM traffic coming to our SmartSwitch for packet processing and transformation. |
+| Upstream Service | The services that programs ENIs to smart switches, e.g., SDN controller. |
+| UD / Update domain | It defines the groups of virtual machines and underlying physical hardware that can be rebooted at the same time. |
+| FD / Fault domain | It defines the group of virtual machines that share a common power source and network switch. |
+| SPOF | Single point of failure. |
+
+For more terminology, please refer to <https://github.com/sonic-net/DASH/wiki/Glossary>.
+
+## 3. Requirements, Assumptions and SLA
+
+### 3.1. Goals
+
+The goal of the SmartSwitch HA design is trying to achieve the following goals:
+
+1. 0 downtime on planned switchover.
+2. <2 sec downtime on failover to standalone setup for each ENI.
+3. Ability to resume connections in the event of both planned and unplanned failover.
+4. After both planned and unplanned failover and recovery, the flow on all DPUs will be aligned.
+5. Only focus on the case, in which only 2 DPUs are in the HA set, although conceptually, we can have more than 2, but it is not in our current scope. 
+6. No complicated hashing logic in switches to ensure packets land on right switch.
```

Comment:
This is a bit ambiguous to me. Do you mean, "No complex hash-based ECMP in T2 to steer packets to correct T1?" or "No complex logic in T1 NPU to steer packet to correct DPU?" or something else?

Reply:
Sure, that's better, thanks,

Reply:
it is updated and pushed now. : D